synapse/tools/snapshot.py

```diff
@@ -0,0 +1,68 @@
+import sys
+import asyncio
+import logging
+import argparse
+
+import synapse.exc as s_exc
+import synapse.telepath as s_telepath
+
+import synapse.lib.output as s_output
+
+logger = logging.getLogger(__name__)
+
+desc = '''
+Command line tool to freeze/resume service operations to allow
+system admins to generate a transactionally consistent volume
+snapshot using 3rd party tools.
+
+The use pattern should be::
+
+    python -m synapse.tools.snapshot freeze
+
+    <generate volume snapshot using 3rd party tools>
+
+    python -m synapse.tools.snapshot resume
+
+The tool will set the process exit code to 0 on success.
+'''
+
+async def main(argv, outp=s_output.stdout):
+
+    pars = argparse.ArgumentParser('synapse.tools.snapshot',
+                        description=desc,
+                        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    subs = pars.add_subparsers(required=True, title='commands', dest='cmd')
```

Comment:
I wonder if a query command would be useful to get the current state of frozen/running? Seems like it should be easy to keep track of the current state but :shrug:

Reply:
I think we can leave that as a future ask :+1: I think it'll be pretty obvious by the combination of paused edits and the warning logs.

Reply:
The paused status is exposed in `getCellInfo()` so it would be easy to add. We'd just need to define the return code values for it as a CLI tool.

Reply:
I did see it in getCellInfo(). That's good enough for me.